Horseshoe-class Cargo Ship

MissyNym opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Added a new ship, the Horseshoe-class Cargo Ship. This is a small ship intended for only 1-2 players, and starts with a very bare-bones set-up, including the following:

- A cargo bay equipped with one set of basic mining gear, an autolathe, one suit storage unit with EVA suit, and one crate with 10 sheets of plasma.
- An engineering bay containing atmos, one SMES, one PACMAN, one tank dispenser equipped with 5 tanks of oxygen, two ion thrusters, and a half-functioning airlock. In case of breaches, there's even a crate fully equipped with a mechanical toolbox and some glass and steel! Driving through ion storms or asteroid fields is still not recommended.
- A barebones crew quarter, equipped with one chair, one table with microwave, two boxes of randomly assorted donk pockets, a cryo unit, and one bed. Toilet and sink with emergency EVA suit included! There's even a carpet for wiping your feet off at the bathroom entrance.
- A cockpit equipped with shutters, holopad, two seats for a pilot and co-pilot, wideband radio, and a glovebox closet containing one double-barrel shotgun, one box each of rubbershot and buckshot, gloves, and a basic medkit.
- Basic atmos piping for proper air recycling and repressurization, in case someone does an oopsie and opens up the cargo bay without using the holo shield. Scrubbing/repressurization still takes time to do- spacers beware!
- A budget with a whopping 1000 credits. Squalorcore activate!

![image](https://github.com/user-attachments/assets/058ab4af-f445-4e50-aedd-a6d4dc9b59bd)

## Why It's Good For The Game

Smaller ships means fragmenting people into smaller crews! The footprint of this is purposefully tiny to try and force the player(s) to interact with planets and other ships more often. Repairing this ship should be easy given the small footprint and relatively compact wiring and atmos setup; the cargo bay is also essentially uncluttered and allows room to specialize the ship into maybe one or two purposes. Most of the ship's space is taken up by the systems keeping it running, so good luck! An ambitious sort might consider expanding this ship's footprint...

## Changelog

:cl:
add: The Horseshoe-class Hauler, a new Independent cargo ship type.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

MissyNym committed:
Created the Hauler-class ship, along with schema file and new independent outfit specific to truckers.

MissyNym committed:
Created mobile docking port. Fingers crossed!

MissyNym committed:
Renamed Hauler to Horseshoe. Also set up the atmos fully with air alarms, configured the SMES and PACMAN setups, and provided some free fuel, one set of basic mining tools, and a pair of fire/EM closet. The airlock is unfortunately still extremely lethal, but that seems to be a trend on other ships, so we'll call it even for now.

Sockssq commented:
gotta add 

* haha funny gimmick ship
because its the perfect ship to run ADMIN or NONADMIN shop gimmicks on due to the cargo space

Sockssq commented:
gotta add 

* haha funny gimmick ship
because its the perfect ship to run ADMIN or NONADMIN shop gimmicks on due to the cargo space

MissyNym commented:
Noting one problem here for future reference: The airlock is hella deadly and doesn't depressurize or repressurize. Anyone who'd like to figure that out, please feel free. Atmos gives me a bit of a headache to manage.

MissyNym committed:
Forgot to finish changes to the .json file, so that would probably cause problems for this actually loading properly, since it's referencing the old map path. Whoops!

MissyNym committed:
Also added in an autolathe into the cargo bay to be nice. Mining materials can't be processed otherwise without returning to an outpost, and I don't want to be THAT mean to the space truckers.

MissyNym committed:
Adjusted engine ship areas a bit. As opposed to one 'Engineering' area, I assigned port and starboard engines-- otherwise there would be two APCs assigned to one area which causes weird interactions with lights and otherwise.

MissyNym committed:
Merge branch 'master' into space-truckin'-ship

MissyNym committed:
Fixed an APC. Linters mentioned a failure due to a banned variable.

MissyNym committed:
Merge branch 'trazodont:master' into space-truckin'-ship

MissyNym committed:
Merge branch 'space-truckin'-ship' of https://github.com/MissyNym/Borktest into space-truckin'-ship

MissyNym committed:
Turns out the atmos wasn't rigged up correctly. Who would've thought? The main air pump should now be active by default. The inside of the ship has also been slightly rearranged to allow enough room for the atmos to be rerouted specifically for the air supply pipes. Truckers should now be able to access their miserable see-through bathroom from both sides, and to eat their microwave meals a little further away from the SMES and generator. Also rearranged the areas a touch.

MissyNym committed:
Turns out the atmos wasn't rigged up correctly. Who would've thought? The main air pump should now be active by default. The inside of the ship has also been slightly rearranged to allow enough room for the atmos to be rerouted specifically for the air supply pipes. Truckers should now be able to access their miserable see-through bathroom from both sides, and to eat their microwave meals a little further away from the SMES and generator. Also rearranged the areas a touch.

MissyNym committed:
Merge branch 'space-truckin'-ship' of https://github.com/MissyNym/Borktest into space-truckin'-ship

MissyNym committed:
Playing musical chairs with the inside. Vent pumps were replaced with high-volume ones as a nicety- if the ship depressurizes, it takes a while to refill otherwise, and I'd like for folks not to die to pressure vacuum that quickly since they don't start off with an EVA suit.

TDHooligan commented:
happy with the idea of merging it, but i'd want a little more just to make it less like a trap if people do use it:
 - add some small living quarters/toilet. i understand it's a small vessel but all the ships have an actual crew bay
 - at the minimum get a couple spacesuits, preferably take some time thinking of other mitigation methods if a breach occurs. like an additional room with a dual atmos firelock (could just be the crew space)
 - add a first aid kit
 - oxygen storage
 - a mechanical toolbox
 - get a 20-stack of metal/glass in there for emergency repairs, since there's little to salvage on board
 - it needs some detail in the cargo hold, as half the ship has functionally no detail. Look at how the other ships use decals.

MissyNym committed:
Added a few more things after review:
- Crew quarters has been separated out from the midship engineering section. This includes a separate toilet with emergency internals closet, as well as a Donk Co. carpet at the entrance to dry your feet on/to make this just a bit more cursed while cutting down on the general level of squalor. God rest ye, one-tile glass doored toilet- you were not meant for this cruel world. The cryo pod has also been moved to this new section. Obligatory Bless This Spess poster added as standard issue.
- The engineering section has been reduced down to plating, with the atmos pipes rerouted. A tank storage has also been added which contains only five oxygen tanks, as opposed to the usual ten. An engineering crate has also been included containing the very basics as suggested; one mechanical toolbox, 20 sheets of steel, 20 sheets of glass.
- Cargo bay has some decals laid down to make it feel a little less barren, as well as some random posters for flavor. Suit storage has also been added to it to give a little more survivability in case space truckers go drifting through an ion storm or a meteor field and need to pull emergency repairs. Good luck with patching the holes, brave drivers.
- Given the toilet's new location, the bridge has a "glove-box" closet with a double-barrel, as well as a box of rubbershot and a box of 12ga to keep our poor truckers alive in hostile environments. First-aid also included!

MissyNym committed:
Forgot an air scrubber for the rearranged engineering bay. Oops. That's been added and the doubled up air vents have been reduced back down to one. Piping situation: Slightly less spaghetti.

TDHooligan commented:
ill double check things this weekend. but it looks like the space-van may be real

MissyNym commented:
so i DID manage to get the airlock working halfway- it'll cycle to the outside just fine, but trying to head back inside don't work so hot. that's half of a victory, but boy if i'm not sure how to solve the other half... also, when cycling to the outside, the external airlocks seem to move about a quarter of a second apart from each other. i'm not sure how much that matters, but it is weird

MissyNym committed:
Merge branch 'master' into space-truckin'-ship

MissyNym committed:
Merge branch 'master' into space-truckin'-ship

TDHooligan commented:
if you can get the airlock workin i'll merge it in whenevs B)
